lib/index.js

```diff
@@ -122,6 +128,37 @@ internals.handler = function (route, options) {
             }
         }
 
-        return responder.view(settings.template, context, settings.options);
+        return h.view(settings.template, context, settings.options);
     };
 };
+
+
+internals.state = (realm) => {
+
+    const state = realm.plugins.vision = realm.plugins.vision || {};
+    return state;
+};
+
+
+internals.nearestState = function (realm) {
+
+    if (realm.plugins.vision) {
+        return realm.plugins.vision;
+    }
+
+    const parent = realm.parent;
+
+    return internals.nearestState(parent);
+};
+
+
+internals.getRootRealm = (realm) => {
+
+    let rootRealm = realm;
```

Comment:
I think `rootRealm` is a slightly odd variable name here– it's not the root realm until it's returned (and not being used as a variable anymore).

Reply:
Agree!